zerver/tests/test_events.py

```diff
@@ -725,130 +725,87 @@ def test_change_full_name(self):
         error = schema_checker('events[0]', events[0])
         self.assert_on_error(error)
 
+    def test_change_message_retention_days(self):
+        # type: () -> None
+        self.do_set_realm_property_test("message_retention_days", [30])
+
     def test_change_realm_name(self):
         # type: () -> None
+        self.do_set_realm_property_test('name', [u'New Realm Name'])
+
+    def do_set_realm_property_test(self, name, values_list):
+        # type: (str, List[Union[int,bool,Text]]) -> None
+
+        property_type = self.user_profile.realm.property_types[name]
+        if property_type is bool:
+            validator = check_bool
+        elif property_type is Text:
+            validator = check_string
+        elif property_type is int:
+            validator = check_int
```

Comment:
I think this should be a fairly complete 

maybe add an `else: raise AssertionError("Unexpected property type %s" % (property_type,))` to provide a clearer error in the case that we add something new and forget to update this code path?